scipy/cluster/tests/test_hierarchy.py

```diff
@@ -812,6 +812,13 @@ def test_valid_orientation(self):
         Z = linkage(hierarchy_test_data.ytdist, 'single')
         assert_raises(ValueError, dendrogram, Z, orientation="foo")
 
+    def test_can_take_label_as_array_or_list(self):
+        Z = linkage(hierarchy_test_data.ytdist, 'single')
+        labels = np.array([1, 3, 2, 6, 4, 5])
```

Comment:
testing with an empty NumPy array might also be useful, though maybe not required

adding a comment to the test like `# test for gh-12418` can be nice too, though again perhaps not required

Reply:
I added a test for empty labels. I am not sure which behaviour we want:

1. an error "Dimensions of Z and labels must be consistent." (this is the current behaviour)
2. `labels=[]` is the same as `labels=None` (`labels=None` is the default)

Reply:
I also added `# test for gh-12418`.

In general I feel `git blame` works reasonably well to find the associated PR with the discussion. At the same time it may be annoying if the history is complicated so I can see why the comment is useful.

Reply:
I'd prefer an error, as it is now.

Reply:
OK great, let me know if you need anything else, maybe a changelog entry somewhere?